app/src/main/java/org/mozilla/tv/firefox/focus/FocusRepo.kt

```diff
@@ -0,0 +1,332 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.focus
+
+import android.view.View
+import android.view.ViewTreeObserver
+import androidx.annotation.VisibleForTesting
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.subjects.BehaviorSubject
+import mozilla.components.browser.engine.system.NestedWebView
+import org.mozilla.tv.firefox.R
+import org.mozilla.tv.firefox.ScreenController
+import org.mozilla.tv.firefox.ScreenControllerStateMachine.ActiveScreen
+import org.mozilla.tv.firefox.pinnedtile.PinnedTileRepo
+import org.mozilla.tv.firefox.pocket.PocketVideoRepo
+import org.mozilla.tv.firefox.session.SessionRepo
+import org.mozilla.tv.firefox.utils.URLs
+
+private const val NESTED_WEB_VIEW_ID = 2147483646 // Int.MAX_VALUE - 1
+
+class FocusRepo(
+    screenController: ScreenController,
+    sessionRepo: SessionRepo,
+    pinnedTileRepo: PinnedTileRepo,
+    pocketRepo: PocketVideoRepo
+) : ViewTreeObserver.OnGlobalFocusChangeListener {
+
+    val defaultFocusMap: HashMap<ActiveScreen, Int> = HashMap()
+
+    data class State(
+        val focusNode: FocusNode,
+        val focused: Boolean = true
```

Comment:
Since we're exposing a few different things from this class (nodes, default focus requests, etc), I find the `State` name a little confusing.  What else could we call this?

Reply:
State no longer exposes default focus requests, I could rename it FocusState if you'd like.